submit.py

```diff
@@ -1,15 +1,27 @@
 #!/usr/bin/env python
-import ConfigParser
+from __future__ import print_function
 import optparse
 import os
 import sys
 import itertools
-import mimetools
 import mimetypes
-import urllib
-import urllib2
 import traceback
-
+import random
+import string
+if sys.version_info[0] >= 3:
+    _PYTHON_VERSION = 3
+    import configparser
+    import urllib.parse
+    import urllib.request
+    import urllib.error
+else:
+    # Python 2, import modules with Python 3 names
+    _PYTHON_VERSION = 2
+    import ConfigParser as configparser
+    import urllib
+    import urllib2
+    urllib.request = urllib.error = urllib2
```

Comment:
Not sure about `urllib2`, but it is definitely not the case that `urllib` provides a superset of `urllib.parse`.  I see your point but then again pretty much any change made will have to be tested in both Python 2 and 3 anyway since there can be subtle differences in behaviour.

With your first suggestion, imports would look like this:
```python
# Py3
from urllib.error import URLError
from urllib.request import Request, HTTPCookieProcessor, build_opener, install_opener, urlopen
from urllib.parse import urlencode

# Py2
from urllib2 import URLError
from urllib2 import Request, HTTPCookieProcessor, build_opener, install_opener, urlopen
from urllib import urlencode
```